pkg/logger/logger.go

```diff
@@ -0,0 +1,328 @@
+package logger
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type Logger struct {
+	l       *zap.Logger
+	pkgs    map[string]bool
+	metrics bool
+
+	LogCount *LogCounter
+}
+
+type LogOrigin struct {
+	File string
+	Line int
+}
+
+type LogCounter struct {
+	rwMutex sync.RWMutex
+	data    map[LogOrigin]uint32
+}
+
+func (lc *LogCounter) update(lo LogOrigin) {
+	lc.rwMutex.Lock()
+	lc.data[lo]++
+	lc.rwMutex.Unlock()
```

Comment:
For best practice, I think you should use the `defer` option for the unlock, to avoid cases in the future while modifying the function that the unlock won't be reached for some reason.
This is true for the following 2 functions as well in my opinion.

Reply:
Just to note: if this becomes a hot path, defer has a performance overhead. But since this is currently only used in debug mode it should be fine.